src/elexsolver/LinearSolver.py

```diff
@@ -29,21 +29,32 @@ class LinearSolver(ABC):
 
     def __init__(self):
         self.coefficients = None
+        self.rng = np.random.default_rng(seed=0)
 
     @classmethod
-    def fit(self, x: np.ndarray, y: np.ndarray, weights: np.ndarray | None = None, lambda_: float = 0.0, *kwargs):
+    def fit(
+        self,
+        x: np.ndarray,
+        y: np.ndarray,
+        weights: np.ndarray | None = None,
+        lambda_: float = 0.0,
+        cache: bool = True,
+        **kwargs,
+    ):
         """
         Fits model
         """
         raise NotImplementedError
 
-    def predict(self, x: np.ndarray) -> np.ndarray:
+    def predict(self, x: np.ndarray, coefficients: np.ndarray | None = None) -> np.ndarray:
```

Comment:
How would you feel about leaving this (public method) like:

```python
def predict(self, x: np.ndarray) -> np.ndarray:
    return _predict(x, self.coefficients)
```

and adding:

```python
def _predict(self, x: np.ndarray, coefficients: np.ndarray) -> np.ndarray:
    return x @ coefficients
```

and then in `residuals()` you call `_predict(x_test, coefficients_k)` ?  The way it's written now invites users to pass in any arbitrary `coefficients`, which might not be a good idea 🤔